[improve][doc] SEO for Pulsar SQL (#677)
apache/pulsar-site: docs/sql-deployment-configurations.md

id: sql-deployment-configurations
title: Pulsar SQL configuration and deployment
sidebar_label: "Configuration and deployment"

> description: Get started to configure the Pulsar Trino plugin and deploy a Pulsar SQL cluster.

Reviewer: ```suggestion
description: Configure the Pulsar Trino plugin and deploy a Pulsar SQL cluster.
```

### Enable authentication and authorization between Pulsar and Pulsar SQL

To enable authentication and authorization between Pulsar and Pulsar SQL, you need to set the following configurations in the `${project.root}/trino/conf/catalog/pulsar.properties` properties file:

```properties
pulsar.authorization-enabled=true
pulsar.broker-binary-service-url=pulsar://localhost:6650
```

:::note
By default, the authentication and authorization between Pulsar and Pulsar SQL are **disabled**.

> :::

Reviewer: Have you previewed this in your local environment?

Reviewer: ?

### Connect Trino to Pulsar with multiple hosts

* To connect Trino with multiple hosts for Pulsar brokers, you need to add multiple URLs to `pulsar.web-service-url`.

> * To connect Trino with multiple hosts for Pulsar ZooKeeper, you need to add multiple URIs to `pulsar.zookeeper-uri`.

Reviewer: I believe the original texts are good. The updated changes are length.

### Connect Trino to Pulsar with multiple hosts

> To configure multiple hosts for brokers, add multiple URLs to `pulsar.web-service-url`.

Reviewer: ```suggestion
To connect Trino with multiple hosts for brokers, add multiple URLs to `pulsar.web-service-url`.
```

> To configure multiple hosts for brokers, add multiple URLs to `pulsar.web-service-url`.

Reviewer: ```suggestion
To connect Trino with multiple hosts for ZooKeeper, add multiple URLs to `pulsar.web-service-url`.
```

### Deploy a cluster on multiple nodes

You can deploy a Pulsar SQL cluster or Trino cluster on multiple nodes. The following steps shows how to deploy a cluster on three-node cluster.

> #### Step 1: Copy the Pulsar binary distribution to three nodes.

Reviewer: ```suggestion
Step 1: Copy the Pulsar binary distribution to three nodes
```

It's not necessary to make them as headings because #### is not shown on the right ToC

discovery.uri=<coordinator-url>
```

> #### step 2: Modify `pulsar.web-service-url` and  `pulsar.zookeeper-uri` configuration in the `${project.root}/trino/conf/catalog/pulsar.properties` file accordingly for the three nodes.

Reviewer: ```suggestion
step 2: Modify `pulsar.web-service-url` and  `pulsar.zookeeper-uri` configuration in the `${project.root}/trino/conf/catalog/pulsar.properties` file accordingly for the three nodes.
```

> #### Step 3: Start the coordinator node:

Reviewer: ```suggestion
Step 3: Start coordinator node
```

```bash
./bin/pulsar sql-worker run
```

> #### Step 4: Start worker nodes:

Reviewer: ```suggestion
Step 4: Start worker nodes
```

```bash
./bin/pulsar sql-worker run
```

> #### Step 5: Start the SQL CLI and check the status of your cluster:

Reviewer: ```suggestion
Step 5: Start the SQL CLI and check the status of your cluster
```

```bash
./bin/pulsar sql --server <coordinate_url>
```

> #### Step 6: Check the status of your nodes:

Reviewer: ```suggestion
Step 6: Check the status of your nodes
```

discovery.uri=<coordinator-url>
```

> step 2: Modify `pulsar.web-service-url` and  `pulsar.zookeeper-uri` configuration in the `${project.root}/trino/conf/catalog/pulsar.properties` file accordingly for the three nodes.

Reviewer: ```suggestion
step 2: Modify `pulsar.web-service-url` and `pulsar.zookeeper-uri` configuration in the `${project.root}/trino/conf/catalog/pulsar.properties` file accordingly for the three nodes.
```

> Step 3: Start the coordinator node:

Reviewer: ```suggestion
Step 3: Start the coordinator node.
```

```bash
./bin/pulsar sql-worker run
```

> Step 4: Start worker nodes:

Reviewer: ```suggestion
Step 4: Start worker nodes.
```

```bash
./bin/pulsar sql-worker run
```

> Step 5: Start the SQL CLI and check the status of your cluster:

Reviewer: ```suggestion
Step 5: Start the SQL CLI and check the status of your cluster.
```

```bash
./bin/pulsar sql --server <coordinate_url>
```

> Step 6: Check the status of your nodes:

Reviewer: ```suggestion
Step 6: Check the status of your nodes.
```
